### PR TITLE
[HOTFIX] make inputDataDir path independant of user

### PR DESCRIPTION
**Pull request by halfway258**
Also added help and more descriptive prints

### CODE REVIEW

File: src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl [hunk 1]
```diff
@@ -98,10 +98,11 @@ idle(cast, {csvList,Workers,CSVData}, State = #source_statem_state{chunkSize = C
 %%This cast spawns a transmitter of data stream towards NerlClient by casting batches of data from parsed csv file given by cowboy source_server
 idle(cast, {startCasting,Body}, State = #source_statem_state{myName = MyName, lengthOfSample = LengthOfSample, sendingMethod = Method, frequency = Frequency, chunkSize = ChunkSize, sourcePid = [],workersMap = WorkersMap, castingTo = CastingTo, nerlnetGraph = NerlnetGraph, msgCounter = Counter, csvName = CSVName, csvList =CSVlist}) ->
     [_Source,NumOfBatchesToSend] = re:split(binary_to_list(Body), ",", [{return, list}]),
+    NumBatches = list_to_integer(NumOfBatchesToSend),
 
   io:format("start casting to: ~p~n, number of batches to send: ~p~ntotal casting list length: ~p~n ",[CastingTo,NumOfBatchesToSend, length(CSVlist)]),
-
-  Transmitter =  spawnTransmitter(CastingTo,CSVName,CSVlist,NerlnetGraph,MyName,WorkersMap,ChunkSize,LengthOfSample,Frequency,list_to_integer(NumOfBatchesToSend),Method) ,
+  NumToSend = if NumBatches < length(CSVlist) -> NumBatches; true -> length(CSVlist) end,
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl:104]: What is CSVlist? 
How are num of batches and csvlist related?
Please add some documentation at this part to explain these questions.

File: src_py/apiServer/apiServer.py [hunk 2]
```diff
@@ -162,9 +162,15 @@ def getQueueData(self):
     def sendDataToSources(self, phase):
         print("***********NOT READY************")
```
Comment (leondavi) [inline src_py/apiServer/apiServer.py:164]: Why to print not ready in any case?

File: src_py/apiServer/experiment.py [hunk 3]
```diff
@@ -40,6 +40,13 @@ def remove0Tails(self):
             for workerRes in csv.workersResList:
                 workerRes.remove0Tail()
 
+    def printExp(self):
+        print(f"""Experiment Data:
+        Data source:    {self.expFlow["CSV path"]}
+        Batches to send per phase:
+            Training:   {self.expFlow[globe.BATHCHES_PER_SOURCE_STR][globe.TRAINING_STR]}
```
Comment (leondavi) [inline src_py/apiServer/experiment.py:47]: BATCHES - typo

File: src_py/apiServer/globalVars.py [hunk 4]
```diff
@@ -26,7 +26,8 @@
 TRAINING_STR = "Training"
 PREDICTION_STR = "Prediction"
 BATHCHES_PER_SOURCE_STR = "Batches per source"
-
+# inputDataPath
+INPUT_DATA_PATH = "/usr/local/lib/nerlnet-lib/NErlNet/inputDataDir/"
```
Comment (leondavi) [inline src_py/apiServer/globalVars.py:30]: Don't we get rid of inputDataDir?

File: src_py/apiServer/transmitter.py [hunk 5]
```diff
@@ -93,9 +93,7 @@ def train(self):
         globe.experiment_flow_global.syncTrainingWithFlow()
 
         # 1 Ack for clientsTraining(), <num of sources> Acks for updateCSV():
-        globe.pendingAcks += 1 + len(globe.components.sources) 
-
-        self.updateCSV(globe.TRAINING_STR)
+        globe.pendingAcks += 1
```
Comment (leondavi) [inline src_py/apiServer/transmitter.py:96]: Please write here a short explanation about what was happened here before

Why did you remove "+ len(globe.components.sources)"

File: src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl [hunk 6]
```diff
@@ -89,7 +89,7 @@ idle(cast, {csvList,Workers,CSVData}, State = #source_statem_state{chunkSize = C
   %%  CSVName = lists:last(re:split(CSVPath,"/",[{return,list}])),
   {RouterHost,RouterPort} = getShortPath(MyName,"mainServer",NerlnetGraph),
   %%  send an ACK to mainserver that the CSV file is ready
-  io:format("source updated transmitting list, total batches to send: - ~p~n",[length(CSVlist)]),
+  io:format("source updated transmitting list, available batches to send: - ~p~n",[length(CSVlist)]),
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl:92]: Try to get use to erlang logger
https://www.erlang.org/doc/man/logger.html

Here you can use info or notice

**Review by leondavi — CHANGES_REQUESTED** [review]

**Comment by leondavi** [comment]
The name of this change misleading.
You actually mean that inputDataDir is changed to an agreed path of /usr/local/lib... which is the same across different linux OS machines. 

Moreover, this PR contains multiple changes that aren't related with the inputDataDir.

File: src_py/apiServer/apiServer.py (hunk 2 above)
Comment (halfway258) [reply src_py/apiServer/apiServer.py]: Changed status

File: src_py/apiServer/transmitter.py (hunk 5 above)
Comment (halfway258) [reply src_py/apiServer/transmitter.py:96]: moved functionability to "sendData" function

File: src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl (hunk 1 above)
Comment (halfway258) [reply src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl:106]: added documentation

File: src_py/apiServer/experiment.py (hunk 3 above)
Comment (halfway258) [reply src_py/apiServer/experiment.py:47]: Fixed

File: src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl [hunk 7]
```diff
@@ -192,7 +195,7 @@ Triplets =getHostPort(WorkersNames,WorkersMap,NerlnetGraph,MyName,[]),
 sendSamples(ListOfSamples,_CSVPath,_ChunkSize,_LengthOfSample, Ms,Pid,_Triplets,Counter,NumOfBatchesToSend,_Method) when NumOfBatchesToSend=<0->
   receive
   after Ms ->
-    gen_statem:cast(Pid,{finishedCasting,Counter,ListOfSamples}), io:format("sent all samples~n")
+    gen_statem:cast(Pid,{finishedCasting,Counter,ListOfSamples}), io:format("sent all samples~n")   %, should exit(normal)?
```
Comment (leondavi) [inline src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl:198]: What is the meaning of comment here?

File: src_py/apiServer/transmitter.py [hunk 8]
```diff
@@ -120,7 +118,7 @@ def contPhase(self, phase):     # phase can be train/training no matter capitals
         print("starting additional training")
 
         globe.pendingAcks += 1
-        if(phase.lower() == "train" or phase.lower() == "training"):
+        if(phase.lower().startswith("t")):      # t for train / training
```
Comment (leondavi) [inline src_py/apiServer/transmitter.py:121]: This is not a good idea to count on a single letter. 
why not startswith "train" which captures both train and training?

File: src_py/apiServer/transmitter.py [hunk 9]
```diff
@@ -147,9 +145,7 @@ def predict(self):
         globe.experiment_flow_global.syncPredicitionWithFlow()
 
         # 1 Ack for clientsPredict(), <num of sources> Acks for updateCSV():
-        globe.pendingAcks += 1 + len(globe.components.sources) 
-
-        self.updateCSV(globe.PREDICTION_STR)
+        globe.pendingAcks += 1
```
Comment (leondavi) [inline src_py/apiServer/transmitter.py:148]: If you can add detailed comment about this ack variable and why it's needed.

**Review by leondavi — CHANGES_REQUESTED** [review]

File: src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl (hunk 7 above)
Comment (halfway258) [reply src_erl/Communication_Layer/http_Nerlserver/src/source/sourceStatem.erl:198]: Small comment I made to myself, changed to describe actual flow

File: src_py/apiServer/transmitter.py (hunk 8 above)
Comment (halfway258) [reply src_py/apiServer/transmitter.py:121]: I just wanted to shorten train / training as much as possible. Doesn't really matter since this is not live user input but API code, changing to startswith("train")

File: src_py/apiServer/transmitter.py (hunk 9 above)
Comment (halfway258) [reply src_py/apiServer/transmitter.py:148]: added

**Review by leondavi — APPROVED** [review]
